common/persistence/visibility/store/elasticsearch/processor.go

```diff
@@ -172,6 +176,15 @@ func (p *processorImpl) hashFn(key interface{}) uint32 {
 
 // Add request to the bulk and return a future object which will receive ack signal when request is processed.
 func (p *processorImpl) Add(request *client.BulkableRequest, visibilityTaskKey string) *future.FutureImpl[bool] {
+	if atomic.LoadInt32(&p.status) == common.DaemonStatusStopped {
```

Comment:
Unfortunately I still don't think this will work. Consider the following case:
1. Status is Started, Add() method passes this check
2. Close() change status to Stopped, Wait() found there's no inflightAdd. and proceed to stop bulk processor and set fields to nil
3. Add() method adds one inflightAdd and continue to process the request.